### ISSUE TITLE
MPavg and MPmin

### ISSUE THREAD

**Issue by huanmofazong** [Openning07/MPADA#5]
Hello, I would like to know which part of the code I should modify if I want to reproduce MPavg and MPmin?

**Comment by Openning07** [comment]
Primarily, you should change the data directory of AVA and make sure the environment is prepared to reproduce the claimed results in my paper. Please read the **Note** and one example code for details. I also upload my checkpoint files in ** MyCheckpoint** folder.

**Comment by huanmofazong** [comment]
@Openning07 Thank you for your reply. I configured the environment and changed the path of the dataset to run your code.
But it should be MPada. I would like to ask which part of the code should be modified if I want to reproduce MPavg and MPmin.

**Comment by Openning07** [comment]
The core difference is the *weights_JensenFactor* in function [compute_loss_and_error](https://github.com/Openning07/MPADA/blob/fd438b26764c1fad2ed4d6d3cc78fdb2c181a970/imagenet_utils.py#L460). If you want to change MPada to other two modes, please change the implementation of *weights_JensenFactor*: for MPavg, *weights_JensenFactor* is a vector [1, 1, ..., 1]; for MPmin, *weights_JensenFactor* is an one-hot vector where 1 indicates the bit of minimum confidence.

**Comment by huanmofazong** [comment]
@Openning07 Thank you for your reply. I would like to ask if the value of --JensenFactor should be set to 0.5 during training?

**Comment by Openning07** [comment]
Yes. I used *0.5* for *JensenFactor* when MPada mode is selected.

**Comment by huanmofazong** [comment]
@Openning07 For MPmin,you said _weights_JensenFactor_ is an one-hot vector where 1 indicates the bit of minimum confidence.May I ask which Tensor I should use, I try to use [target_p](https://github.com/Openning07/MPADA/blob/fd438b26764c1fad2ed4d6d3cc78fdb2c181a970/imagenet_utils.py#L472)，But that doesn't seem right.
`weights_JensenFactor_old = tf.zeros_like(target_p, dtype='float32')

weight_index = tf.argmin(target_p, axis=0)

index = tf.cast(weight_index, dtype='int32')

weight_a = weights_JensenFactor_old[:index]

weight_b = tf.constant([1.], dtype='float32')

weight_c = weights_JensenFactor_old[index+1:]

weights_JensenFactor = tf.concat([weight_a, weight_b, weight_c], axis=0)`

Is this the right _weights_JensenFactor_?Thank you for your reply.
